src/settingsMigrator.ts

```diff
@@ -1,7 +1,12 @@
-import {TodoistSettings} from "./DefaultSettings";
+import {TodoistSettings, keywordTodoistQuery} from "./DefaultSettings";
 
 export function migrateSettings(settings: any) : TodoistSettings {
 	let newSettings : any = settings;
+
+	if (getSettingsVersion(newSettings) == 1) {
```

Comment:
Yep, you're right.  Seems like I was trying to be clever but missed something obvious.  I did the quick and dirty version to get it fixed in the short term.

I'm going to chew on it about a little bit about how to better approach it.  Will update here on if I'm going to attack it for this PR, or weasel out and defer it until later.  :)

Reply:
I didn't forget about this!  I haven't figured out how I want to attack it, either.  Since we're hopefully getting close to being feature complete on this PR, it's your call on if you want to wait for inspiration to strike.

Reply:
let's exclude it from this PR! esp. since someone else filed https://github.com/wesmoncrief/obsidian-todoist-text/issues/25